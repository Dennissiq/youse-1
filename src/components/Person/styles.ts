import styled from 'styled-components'

export const Div = styled.div`
  padding: 30px 20px;
  background-color: ${({ modifier }: { modifier: number }) =>
    modifier % 2 === 0 ? '#6a418c' : '#7d3caf'};
  
  transition: transform .75s ease-in-out;
  
  &.enter {
    ${({ modifier }: { modifier: number }) => modifier % 2 === 0 ? `
    transform: translateX(-100vw);
    ` : `
    transform: translateX(100vw);
    `}

    &.header {
      opcaity: 0;
      transition: opacity .5s linear
    }

    article {
      opacity: 0;
    }
  }

  &.header {
    transform: translateX(0);
    opacity: 1;
  }

  position: relative;
  transform: translateX(0);

  article {
    transition: opacity .5s .5s linear;
    opacity: 1;
  }
`

export const Article = styled.article`
  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: 600px;
  margin: 0 auto;
`

export const H3 = styled.h3`
  font-size: 16px;
  ${({ header }: { header: boolean }) =>
    header ? `font-weight: bold;` : `font-weight: normal; color: #fff;`};
  @media (max-width: 320px) {
    font-size: 12px;
  }
`

export const Span = styled.span`
  ${({ header }: { header: boolean }) =>
    header
      ? `font-size: 16px;
  font-weight: bold;`
      : `font-size: 14px;
  font-weight: normal;
  color: #fff;`};
  @media (max-width: 320px) {
    font-size: 10px;
  }
`
