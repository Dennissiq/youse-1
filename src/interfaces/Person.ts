export default interface Person {
  id: number
  title: string
  description: string
}
