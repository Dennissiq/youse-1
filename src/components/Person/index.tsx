import React from 'react'
import Person from '../../interfaces/Person'
import { Div, Article, H3, Span } from './styles'

export default ({
  item,
  modifier,
  header,
}: {
  item: Person
  modifier: number
  header: boolean
}) => (
  <Div className={header ? 'header' : ''} modifier={modifier}>
    <Article>
      <H3 header={header}>{item.title}</H3>
      <Span header={header}>{item.description}</Span>
    </Article>
  </Div>
)
