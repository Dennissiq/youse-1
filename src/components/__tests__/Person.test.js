import React from 'react'
import renderer from 'react-test-renderer'
import Person from '../Person'

it('renders correctly', () => {
  const tree = renderer
    .create(<Person item={{ title: 'Mock title', description: 'Mock desc' }} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
