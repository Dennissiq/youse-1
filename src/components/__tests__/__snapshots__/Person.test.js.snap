// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdnylx hmsItK"
>
  <article
    className="sc-gtssRu dLuDwB"
  >
    <h3
      className="sc-dlnjPT ibOGxv"
    >
      Mock title
    </h3>
    <span
      className="sc-hKFyIo beMXmz"
    >
      Mock desc
    </span>
  </article>
</div>
`;
